flexmeasures/data/models/data_sources.py

```diff
@@ -427,39 +426,48 @@ def keep_latest_version(
         event_column = "event_end"
 
     # Add source-related columns using vectorized operations for clarity
+    source_values = set(bdf.index.get_level_values("source").values)
     source_to_fields = {
         s: {
             "source.name": s.name,
             "source.type": s.type,
             "source.model": s.model,
             "source.version": Version(s.version or "0.0.0"),
         }
-        for s in bdf["source"].unique()
+        for s in source_values
     }
-    source_expanded = bdf["source"].map(source_to_fields)
+    source_expanded = bdf.index.get_level_values("source").map(source_to_fields)
+
     bdf[["source.name", "source.type", "source.model", "source.version"]] = (
         pd.DataFrame(source_expanded.tolist(), index=bdf.index)
     )
-
+    bdf["_" + event_column] = bdf.index.get_level_values(event_column)
     # Sort by event_start and version, keeping only the latest version
-    bdf = bdf.sort_values(by=[event_column, "source.version"], ascending=[True, False])
+    bdf = bdf.sort_values(
+        by=["_" + event_column, "source.version"], ascending=[True, False]
+    )
 
     # Drop duplicates based on event_start and source identifiers, keeping the latest version
     unique_columns = [
-        event_column,
-        "cumulative_probability",
```

Comment:
Afaik we're not testing whether we're correctly getting the latest version in case of probabilistic data. I have trouble judging the effect of this removal.